lib/gems/pending/util/object_storage/miq_ftp_storage.rb

```diff
@@ -0,0 +1,92 @@
+require 'util/miq_ftp_lib'
+require 'util/miq_object_storage'
+require 'logger'
+
+class MiqFtpStorage < MiqObjectStorage
```

Comment:
Can we move the existing implementation of `.new_with_opts` from `MiqObjectStorage` to this class? It's a bit strange right now that it is defined in the base class and we have two usable sub-classes, one of which overrides it. I feel like it would be more clear if each class defined it itself unless you know of a third which would share the implementation with `MiqFtpStorage`?